File: variants/java-maven2/net.pijcke.nim.humanconsole/src/main/java/net/pijcke/nim/humanconsole/ConsoleHumanFactory.java
package net.pijcke.nim.humanconsole;

import net.pijcke.nim.rules.Player;
import net.pijcke.nim.rules.PlayerFactory;

import java.util.Scanner;

public class ConsoleHumanFactory implements PlayerFactory {

    private static Scanner scanner = new Scanner(System.in);

    @Override
    public String label() {
        return "Human - Console";
    }

    @Override
    public Player build() {
        System.out.println("Please enter your name : ");
        String name = scanner.nextLine();
        return new ConsoleHuman(name);
    }
}

File: variants/kotlin-gradle/nim/humanconsole/src/main/kotlin/net/pijcke/nim/humanconsole/ConsoleHumanFactory.kt
package net.pijcke.nim.humanconsole

import net.pijcke.nim.rules.Player
import net.pijcke.nim.rules.PlayerFactory
import java.util.*

class ConsoleHumanFactory : PlayerFactory {

    private val scanner = Scanner(System.`in`)

    override fun label() = "Human : Console"

    override fun build() : Player {
        println("Please enter player's name : ")
        val name = scanner.nextLine()
        return ConsoleHuman(name)
    }

}